src/main/scala/io/iohk/ethereum/blockchain/sync/PeerListSupport.scala

```diff
@@ -50,12 +50,11 @@ trait PeerListSupport {
       removePeer(peerId)
   }
 
-  def peerById(peerId: PeerId): Option[Peer] = handshakedPeers collectFirst {
+  def getHandshakedPeerById(peerId: PeerId): Option[Peer] = handshakedPeers collectFirst {
     case (peer, _) if peer.id == peerId => peer
   }
 
-  def blacklistIfHandshaked(peer: Peer, reason: String): Unit =
-    handshakedPeers.get(peer).foreach(_ => blacklist(peer.id, syncConfig.blacklistDuration, reason))
+  def blacklistPeer(peer: Peer, reason: String): Unit = blacklist(peer.id, syncConfig.blacklistDuration, reason)
```

Comment:
I have removed the double call to `handshakedPeers.get(peer)`

Reply:
I tend to see a lot of redundant code, the same checks done over and over again, and this is one of them

Reply:
Removing `handshakedPeers.get(peer).foreach` changes the behavior of the method though. The point of this method is to only blacklist when the peer is handshaked, otherwise we could use `blacklist` directly in the actor.

So we shouldn't rename the method or remove the check.

Reply:
This method is called right after a call to `handshakedPeers.get(peer)`, that it what I meant by redundant code, and it's only called from a single place.